prereise/gather/demanddata/eia/map_ba.py

```diff
@@ -95,3 +95,32 @@ def map_buses_to_county(bus_county_map):
         except TypeError:
             bus_no_county_match.append(index)
     return bus_county_map, bus_no_county_match
+
+
+def map_buses_to_ba(bus_gis):
+    """Find the Balancing Authority in the U.S. territory that each bus in the query grid
+    belongs to.
+
+    :param (*pandas.DataFrame*) bus_gis: data frame contains a list of
+        entries with lat and long of buses.
+    :return: (*pandas.DataFrame*) data frame of BA that buses
+        locate
+    """
+
+    b2c_key = map_buses_to_county(bus_gis)[0].iloc[:, :-1].dropna()
```

Comment:
These are ignored because we want the calculated metrics to be valid (not NaN values) so that it can be compared to the same bus to BA metrics from Plexos system. I will add this explanation on documentation

Reply:
Do you mean write to a csv file as a return value? and put it under data folder?

Reply:
resolved by changes the matching algorithm as suggested and add explanation for possible NaN values in the function return.